### PR TITLE
427 consolidate existing and new

### LINKED ISSUE #427: Consolidate new frontend setup with existing code
- Remove nx, we're using lerna

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #427

## Description

- Moves data loader and data service within the new monorepo pattern (i.e. with lerna)
- Use ts-node to build/execute the data-loader
- Use ts-node with nodemon to build and serve the data-service in watch mode
- Adds build and test steps to CI (fixes the failing CI builds)

Key points to review would be the scripts - both do they work and do the naming conventions work ok for you?

There are Dockerfiles within the `unified-codes` folder to build an image of the data-service, which appear to have been used in CI at one point. The CI has been broken for a while (I can only see 12 months of GH actions history but it was failing in the same way at least then) which leads me to believe these images perhaps aren't used anymore? I'll find out about this before I either update or delete them :)

The `unified-codes` folder, which uses the `nx` setup, still remains, but now only contains the existing web UI. This will need to stick around at least until we reimplement existing functionality in the new frontend.

Next will look to merge this stuff into #425 and get those tests running on `yarn test` as expected

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,71 @@
...
 
-1. Setup Dgraph (see instructions [here](unified-codes/tools/scripts/dgraph/README.md)).
-3. Populate graph schema and demo data: `npx nx serve data-loader`.
-4. Dgraph is now running, you can access at [http://localhost:9080](http://localhost:9080).
+- Install dependencies (Using node v20+ and yarn):
+  - `yarn install`
+- Setup Dgraph (see instructions [here](unified-codes/tools/scripts/dgraph/README.md))
+  - You can then use the [explorer](https://play.dgraph.io/?latest) (server URL is http://localhost:8080)
+- Load data into DGraph:
+  - `yarn load-data`
```
Comment (jmbrunskill) [inline README.md:18]: Any reason we shouldn't combine `load-data` and `update-dgraph-schema`.
I guess it's good having both options...

File: README.md [hunk 2]
```diff
@@ -1,58 +1,71 @@
...
-### data-service
+### Frontend
 
-Backend GraphQL server for accessing product data.
+We are currently re-implementing the frontend. For the existing frontend:
 
-1. Run GraphQL server: `npx nx serve data-service`.
-2. Access v1 api at http://localhost:4000/v1/graphql  
-3. Access v2 api at http://localhost:4000/v2/graphql
+1. Run web server: `npx nx serve web`
```
Comment (jmbrunskill) [inline README.md:30]: I get this. Not sure it's a big deal though. Are you expecting to to work with this branch though?
```npx nx serve web

 >  NX   The current directory isn't part of an Nx workspace.

   To create a workspace run:
   npx create-nx-workspace@latest <workspace name>
   
   To add Nx to an existing workspace with a workspace-specific nx.json, run:
   npx nx@latest init


 >  NX   For more information please visit https://nx.dev/```

File: README.md (hunk 1 above)
Comment (jmbrunskill) [inline README.md:18]: ```@universal-codes/data-loader: [Error: ENOENT: no such file or directory, open '/Users/jamesbrunskill/unified-codes/data/v2/schema.gql'] {```

File: README.md [hunk 3]
```diff
@@ -1,58 +1,71 @@
...
 
-Reusable types and interfaces used by both frontend and backend code.
+- Generate GraphQL Types:
+  - `yarn generate`
 
-### ui
+## Build and release
 
-Reusable frontend React components. Component grouping based on [material-ui][https://material-ui.com/] and [atomic design](https://bradfrost.com/blog/post/atomic-web-design/) principles.
+There was the option of running the data service within Docker - this hasn't been updated with the code rearrange, going to wait to hear if it is being used first! We were building a Docker image in CI, but this has been broken for a long time, so have taken this out for now.
```
Comment (jmbrunskill) [inline README.md:50]: Can be removed now...

**Review by jmbrunskill — APPROVED** [review]
Added a couple of small comments but merge away when you're ready.

File: README.md (hunk 3 above)
Comment (lache-melvin) [reply README.md]: [45f8e61](https://github.com/msupply-foundation/unified-codes/pull/430/commits/45f8e6170810fac44545c6cd00584cf9b8279e15)

File: README.md (hunk 2 above)
Comment (lache-melvin) [reply README.md]: [9bc2942](https://github.com/msupply-foundation/unified-codes/pull/430/commits/9bc29428a278778c35cdbc8af8ee2cb494201a83)

File: README.md (hunk 1 above)
Comment (lache-melvin) [reply README.md:18]: [8a09673](https://github.com/msupply-foundation/unified-codes/pull/430/commits/8a09673f1dee99c9767c0151cbddfe15d7716252)

Comment (lache-melvin) [reply README.md:18]: It's a good question... I think having the separate update-dgraph-schema is useful, if we don't want to a full reload of the data, but we should always run it after the load... will update script now 🚀

Comment (lache-melvin) [reply README.md:18]: [82b0f35](https://github.com/msupply-foundation/unified-codes/pull/430/commits/82b0f35feb23dee42ca027d1404f8858ba254052) ... we'll need to come back to this whole data-loader piece... its probably quite useful from a development standpoint, but we'll phase it out for production, and we'll want to review how we manage the schema then too

Comment (jmbrunskill) [reply README.md:18]: Makes a lot of sense.

**Pull request merged**